File: app/src/main/java/com/OdiousPanda/rainbow/API/UnsplashCall.java
package com.OdiousPanda.rainbow.API;

import com.OdiousPanda.rainbow.DataModel.Unsplash.Unsplash;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface UnsplashCall {
    @GET("photos/random/")
    Call<Unsplash> getRandom(@Query("client_id") String key, @Query("orientation") String orientation);

    @GET("photos/random/")
    Call<Unsplash> getRandomWeather(@Query("client_id") String key, @Query("orientation") String orientation, @Query("query") String query);

}

File: app/src/main/java/com/odiousPanda/rainbowKt/apis/UnsplashCall.kt
package com.odiousPanda.rainbowKt.apis

import com.odiousPanda.rainbowKt.model.dataSource.unsplash.Unsplash
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface UnsplashCall {
    @GET("photos/random/")
    suspend fun getRandom(
        @Query("client_id") key: String = APIConstants.UNSPLASH_KEY,
        @Query("orientation") orientation: String = APIConstants.UNSPLASH_ORIENTATION_PORTRAIT
    ): Response<Unsplash>

    @GET("photos/random/")
    suspend fun getRandomWeather(
        @Query("client_id") key: String = APIConstants.UNSPLASH_KEY,
        @Query("orientation") orientation: String = APIConstants.UNSPLASH_ORIENTATION_PORTRAIT,
        @Query("query") query: String?
    ): Response<Unsplash>

}